pkg/steps/multi_stage.go

```diff
@@ -276,9 +276,16 @@ func (s *multiStageTestStep) createCredentials() error {
 			if err != nil {
 				return fmt.Errorf("could not read source credential: %v", err)
 			}
-			raw.Namespace = s.jobSpec.Namespace
-			raw.Name = name
-			toCreate[name] = raw
+			toCreate[name] = &coreapi.Secret{
```

Comment:
You could also just replace the embedded ObjectMeta, then you don't have to correctly copy over all fields

Reply:
You say potato, I say potato